src/hub/HubRegistry.sol

```diff
@@ -108,4 +108,9 @@ contract HubRegistry is Auth, IHubRegistry {
     function isRegistered(AssetId assetId) public view returns (bool) {
         return _decimals[assetId] != 0;
     }
+
+    /// @inheritdoc IHubRegistry
+    function unitAmount(AssetId assetId) external view returns (uint128) {
+        return (10 ** this.decimals(assetId.raw())).toUint128();
```

Comment:
You can make decimals `public` to avoid the `this` (which implies calling itself like an external)

Reply:
Do we really need `unitAmount()`, it seems to be called only once in the Hub

Reply:
Good point, can be removed